dev-tools/reconfigurator-cli/src/main.rs

```diff
@@ -790,15 +880,30 @@ fn cmd_blueprint_diff_inventory(
     let collection = sim.collections.get(&collection_id).ok_or_else(|| {
         anyhow!("no such inventory collection: {}", collection_id)
     })?;
-    let blueprint = sim
-        .blueprints
-        .get(&blueprint_id)
-        .ok_or_else(|| anyhow!("no such blueprint: {}", blueprint_id))?;
-
+    let blueprint = sim.blueprint_lookup(blueprint_id)?;
     let diff = blueprint.diff_sleds_from_collection(&collection);
     Ok(Some(diff.display().to_string()))
 }
 
+fn cmd_blueprint_save(
+    sim: &mut ReconfiguratorSim,
+    args: BlueprintSaveArgs,
+) -> anyhow::Result<Option<String>> {
+    let blueprint_id = args.blueprint_id;
+    let blueprint = sim.blueprint_lookup(blueprint_id)?;
+
+    let output_path = &args.filename;
+    let outfile = std::fs::OpenOptions::new()
+        .create_new(true)
+        .write(true)
+        .open(output_path)
+        .with_context(|| format!("open {:?}", output_path))?;
+    serde_json::to_writer_pretty(&outfile, &blueprint)
+        .with_context(|| format!("writing to {:?}", output_path))
+        .unwrap_or_else(|e| panic!("{:#}", e));
+    Ok(Some(format!("saved blueprint {} to {:?}", blueprint_id, output_path)))
```

Comment:
Same here:

```suggestion
    let outfile = std::fs::OpenOptions::new()
        .create_new(true)
        .write(true)
        .open(output_path)
        .with_context(|| format!("open {}", output_path))?;
    serde_json::to_writer_pretty(&outfile, &blueprint)
        .with_context(|| format!("writing to {}", output_path))
        .unwrap_or_else(|e| panic!("{:#}", e));
    Ok(Some(format!("saved blueprint {} to {}", blueprint_id, output_path)))
```

And just like `from_reader`, `to_writer{,_pretty}` will also write one byte at a time. Need a `BufWriter` here, or just buffer fully into a string before writing it.